### PR TITLE
emacs: build pgtk variant with full image support

### PR DESCRIPTION
**Pull request by thblt**
###### Description of changes

Enabled png, jpg, … etc support in pgtk Emacs. This was guarded by (withX || withNS), which was false for wayland pgtk 

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by thblt** [comment]
Sorry for the noise, but tagging maintainers just to ensure this gets noticed :) @lovek323 @jwiegley @adisbladis @matthewbauer @atemu

File: pkgs/applications/editors/emacs/generic.nix
Comment (Atemu) [inline pkgs/applications/editors/emacs/generic.nix:196]: Please drop these unrelated formatting changes.

File: pkgs/applications/editors/emacs/generic.nix
Comment (Atemu) [inline pkgs/applications/editors/emacs/generic.nix:148]:
```suggestion
      [ xlibsWrapper libXaw Xaw3d gconf cairo ]
```

File: pkgs/applications/editors/emacs/generic.nix
Comment (Atemu) [inline pkgs/applications/editors/emacs/generic.nix:149]:
```suggestion
    ++ lib.optionals (withX || withPgtk)
```

File: pkgs/applications/editors/emacs/generic.nix
Comment (Atemu) [inline pkgs/applications/editors/emacs/generic.nix:150]: What is libXpm for? That sounds like an x11-specific library, no?

**Review by Atemu — CHANGES_REQUESTED** [review]

**Comment by Atemu** [comment]
Nit: A little explanation in your commit message would be nice.

File: pkgs/applications/editors/emacs/generic.nix
Comment (thblt) [reply pkgs/applications/editors/emacs/generic.nix:150]: I understand [it's for parsing the XPM image format](https://gitlab.freedesktop.org/xorg/lib/libxpm). NixOS packages it under `xorg.`, but it's not X-specific.

**Comment by thblt** [comment]
Thanks for your review! I've extended the commit message with a short explanation of the changes.

File: pkgs/applications/editors/emacs/generic.nix
Comment (thblt) [reply pkgs/applications/editors/emacs/generic.nix]: Erm, sorry about those. Fixed.

File: pkgs/applications/editors/emacs/generic.nix
Comment (Atemu) [inline pkgs/applications/editors/emacs/generic.nix:150]: Does NS really not need libsrvg? If not, we can merge these.

**Review by Atemu — APPROVED** [review]

File: pkgs/applications/editors/emacs/generic.nix
Comment (thblt) [reply pkgs/applications/editors/emacs/generic.nix:150]: This library was already excluded from the NS build. I don't use a Mac, so I can't do any test in that regard. Maybe this should be left for another PR?

**Comment by thblt** [comment]
Please don't merge right now, I forgot something (and did stupid tests so I didn't notice it).  It should just be an extra change.

**Comment by thblt** [comment]
This time it should be good. I've pushed it as a fixup commit so it's easier to review, I'll squash when it's done! I'm building X Emacs again just to make sure it works correctly in all configs, will report.

**Comment by thblt** [comment]
> I'm building X Emacs again just to make sure it works correctly in all configs, will report.

Done. Pgtk, X, and pure terminal emacs build and work correctly. Both X and Pgtk can decode images.

File: pkgs/applications/editors/emacs/generic.nix
Comment (Atemu) [inline pkgs/applications/editors/emacs/generic.nix:183]: This is a sensible change I think but is it important for image support?

**Review by Atemu — APPROVED** [review]

File: pkgs/applications/editors/emacs/generic.nix
Comment (thblt) [reply pkgs/applications/editors/emacs/generic.nix]: Fixed.  `--with-cairo` is implied.

**Comment by thblt** [comment]
Squashed everything, this is the same as the previous tree except for the unnecessary `--with-cairo`

File: pkgs/applications/editors/emacs/generic.nix
Comment (thblt) [reply pkgs/applications/editors/emacs/generic.nix]: Just to be clear the rest of the change is to avoid the final `else` that's meant for pure terminal Emacs and will add `--with-png=no --with-jpeg=no` and so on.

**Comment by thblt** [comment]
I believe this can be merged.

**Pull request merged**
